ImageNet/inference/IluvatarCoreX-P.S.R_resnet50_1p4_tensorflow.json

```diff
@@ -0,0 +1,18 @@
+{
+    "version": "v1.0",
+    "author": "Iluvatar CoreX, P.S.R team: Gang Xu, Yu Song, Tao Yang, Fanwu Han",
+    "authorEmail": "gang.xu@iluvatar.ai",
+    "framework": "TensorFlow 1.13.1",
+    "codeURL": "https://github.com/gnuwind/dawnbench_inference_imagenet",
+    "commitHash": "6e05a35d1f431c697bd421117b8d0a2c594ac725",
+    "model": "ResNet50",
+    "hardware": "1 P4 / 128 GB / 32 CPU",
+    "latency": 1.868,
```

Comment:
@gnuwind what batch size did you use? The [default](https://github.com/gnuwind/dawnbench_inference_imagenet/blob/6e05a35d1f431c697bd421117b8d0a2c594ac725/inference.py#L245) is 8 which wouldn't qualify for the inference latency category. You must use a batch size of 1.

Reply:
@codyaustun Thank you for your review. We specified batch_size=1 by option when we ran it. I just updated [README.md](https://github.com/gnuwind/dawnbench_inference_imagenet/blob/master/README.md) and added the command line we used.

Reply:
Can't reproduce here.When set batch_size=1, it print python: regionFormat.cpp:56: size_t nvinfer1::RegionFormatB::memorySize(int, const nvinfer1::Dims&) const: Assertion `batchSize > 0' failed.
Aborted (core dumped)

Reply:
Can't reproduce here.When I set batch_size=8, the latency is 36.9958ms, when set batch_size = 1, it prints as follows, 
python: regionFormat.cpp:56: size_t nvinfer1::RegionFormatB::memorySize(int, const nvinfer1::Dims&) const: Assertion `batchSize > 0' failed.
Aborted (core dumped)
From the result of batch_size=8, we may infer that latency should be larger than 4.6ms
My environment is 1P4, 16u, 32G.

Reply:
@rankeey Have you used the same container mentioned in our submission json file to run it?  If not, please try it, hope this will help you.

Reply:
@gnuwind the submission looks okay to me. I'll assume that your suggestion solved the issue @rankeey raised because he/she hasn't responded. If @rankeey there is a problem, please raise an issue and we can revisit this pull request. If I don't hear anything in 24 hours, I'll accept and merge in this pull request. Thanks for your patience.

Reply:
@rankeey Thank you for your interest in our result.
Let me confirm the details about your environment and then explain where the gap between our result and your result comes from.
1) The environment we run is a physical server(located on our intranet). Not sure if your environment is a physical server or cloud VM. If it is the latter, virtualization will bring some overhead.

2) I noticed that you mentioned that you used 8 cores. Even if you change the options (taskset, inter)  to match the number of cores, there are still some performance degradations due to the reduction in the number of cores. By the way, our CPU is Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz (Skylake)

3) Our difference is not only CPU/MEM, but also OS performance optimizations. Our environment is a performance-optimized OS that is part of our product. These optimizations for inference performance allow the software to use hardware resources more efficiently, thus reducing our latency. The improvement brought by these optimizations is larger than imagined. I think this may be the prime reason for the gap between our result and yours. We believe that whether it is inference or training, the industry is not able to discuss performance separately from the specific software optimization. So every product or service provider has made the overall optimization to get better performance.

Reply:
To avoid data transfer and augmentation delay, I directly use data produced in memory like this:
    def model_fn(features, labels, mode):
        del features
        del labels
        features = tf.random_normal(shape=[batch_size, 224, 224, 3], dtype=tf.float32)
        labels = tf.random_uniform(maxval=999, minval=0, shape=[batch_size], dtype=tf.int32)


        logits_out = tf.import_graph_def(frozen_graph,
                                         input_map={'input_tensor': features},
                                         return_elements=['softmax_tensor:0'],
                                         name='')

        predictions = {
            'probabilities': logits_out[0],
        }

        if mode == tf.estimator.ModeKeys.PREDICT:
            return tf.estimator.EstimatorSpec(
                mode,
                predictions=predictions)

And the result is as follows,  
    step 1000/6250, iter_time(ms)=2.3601, images/sec=3389
    step 2000/6250, iter_time(ms)=2.2736, images/sec=3518
    step 3000/6250, iter_time(ms)=2.3859, images/sec=3353
    step 4000/6250, iter_time(ms)=2.3205, images/sec=3447
    step 5000/6250, iter_time(ms)=2.2554, images/sec=3546
    step 6000/6250, iter_time(ms)=2.2712, images/sec=3522
    step 7000/6250, iter_time(ms)=2.3422, images/sec=3415
    step 8000/6250, iter_time(ms)=2.3177, images/sec=3451
    step 9000/6250, iter_time(ms)=2.3246, images/sec=3441
    step 10000/6250, iter_time(ms)=2.3026, images/sec=3474
    step 11000/6250, iter_time(ms)=2.3277, images/sec=3436
    step 12000/6250, iter_time(ms)=2.3665, images/sec=3380
    step 13000/6250, iter_time(ms)=2.3417, images/sec=3416
    step 14000/6250, iter_time(ms)=2.3575, images/sec=3393
    step 15000/6250, iter_time(ms)=2.4502, images/sec=3265
    step 16000/6250, iter_time(ms)=2.2862, images/sec=3499
    step 17000/6250, iter_time(ms)=2.3277, images/sec=3436
    step 18000/6250, iter_time(ms)=2.3007, images/sec=3477
    step 19000/6250, iter_time(ms)=2.2702, images/sec=3523
    step 20000/6250, iter_time(ms)=2.3475, images/sec=3407
    step 21000/6250, iter_time(ms)=2.3108, images/sec=3462
    step 22000/6250, iter_time(ms)=2.2669, images/sec=3529
    step 23000/6250, iter_time(ms)=2.3317, images/sec=3430
    step 24000/6250, iter_time(ms)=2.2783, images/sec=3511
    step 25000/6250, iter_time(ms)=2.2840, images/sec=3502
    step 26000/6250, iter_time(ms)=2.3134, images/sec=3458
    step 27000/6250, iter_time(ms)=2.2981, images/sec=3481
    step 28000/6250, iter_time(ms)=2.4114, images/sec=3317
    step 29000/6250, iter_time(ms)=2.3425, images/sec=3415
    step 30000/6250, iter_time(ms)=2.3258, images/sec=3439
    step 31000/6250, iter_time(ms)=2.2759, images/sec=3515
    step 32000/6250, iter_time(ms)=2.3489, images/sec=3405
    step 33000/6250, iter_time(ms)=2.3096, images/sec=3463
    step 34000/6250, iter_time(ms)=2.2979, images/sec=3481
    step 35000/6250, iter_time(ms)=2.4211, images/sec=3304
    step 36000/6250, iter_time(ms)=2.2759, images/sec=3515
    step 37000/6250, iter_time(ms)=2.2993, images/sec=3479
    step 38000/6250, iter_time(ms)=2.4254, images/sec=3298
    step 39000/6250, iter_time(ms)=2.3315, images/sec=3431
    step 40000/6250, iter_time(ms)=2.3680, images/sec=3378
    step 41000/6250, iter_time(ms)=2.3022, images/sec=3474
    step 42000/6250, iter_time(ms)=2.2874, images/sec=3497
    step 43000/6250, iter_time(ms)=2.3198, images/sec=3448
    step 44000/6250, iter_time(ms)=2.3572, images/sec=3393
    step 45000/6250, iter_time(ms)=2.3067, images/sec=3468
    step 46000/6250, iter_time(ms)=2.4302, images/sec=3291
    step 47000/6250, iter_time(ms)=2.3336, images/sec=3428
    step 48000/6250, iter_time(ms)=2.3322, images/sec=3430
    step 49000/6250, iter_time(ms)=2.3191, images/sec=3449
    step 50000/6250, iter_time(ms)=2.2967, images/sec=3483
calculating accuracy...
WARNING:tensorflow:From inference.py:314: map_and_batch (from tensorflow.contrib.data.python.ops.batching) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.data.experimental.map_and_batch(...)`.
results of ./resnet50_imagenet_int8.pb:
    top5 accuracy: 0.51
    latency_mean(ms): 2.3266

And the gpu util is about 81%, I think it is the limitation when batch_size = 1, can you tell me what gpu util you can reach?

Reply:
> 3. Our difference is not only CPU/MEM, but also OS performance optimizations. Our environment is a performance-optimized OS that is part of our product. These optimizations for inference performance allow the software to use hardware resources more efficiently, thus reducing our latency. The improvement brought by these optimizations is larger than imagined. I think this may be the prime reason for the gap between our result and yours. We believe that whether it is inference or training, the industry is not able to discuss performance separately from the specific software optimization. So every product or service provider has made the overall optimization to get better performance.

@gnuwind Thanks for giving more detail. Can you make a note of this in your JSON file under `misc`? Assuming it is an on-premise solution, I would list something like `"OS": "Iluvatar Corex Proprietary OS"`. If it is publically available, include another field with a link.

Reply:
@codyaustun The OS info has been added under `misc`, please refer to the updated version. Thanks for your review.